src/libsystemd-network/dhcp-option.c

```diff
@@ -48,6 +49,35 @@ static int option_append(uint8_t options[], size_t size, size_t *offset,
                 *offset += 1;
                 break;
 
+        case SD_DHCP_OPTION_USER_CLASS: {
+                _cleanup_strv_free_ char **l = NULL;
+                unsigned i = 0;
+                char **s;
+
+                l = strv_split(optval, " ");
```

Comment:
oom check missing

Reply:
also, what's the charset of these strings? and are embedded whitespace allowed? if so, then we should honour ""-enclosed strings. i.e. strv_split_extract() or so should be used